lib/job-storage/firestore-job-storage.js

```diff
@@ -106,4 +106,19 @@ async function removeParent(parentCorrelationId) {
   }
 }
 
-export { storeParent, completedChild, parentIsComplete, removeParent };
+async function messageAlreadySeen(idempotencyKey, deliveryAttempt) {
+  try {
+    await db
+      .collection("idempotencyLocks")
+      .doc(`${idempotencyKey}:${deliveryAttempt}`)
+      .create({ idempotencyKey, deliveryAttempt });
```

Comment:
Vi bör ha även en TTL fält här som är en [timestamp](https://cloud.google.com/firestore/docs/ttl?_ga=2.86947608.-1216045325.1685022633#ttl_fields_and_indexes), så att vi senare kan lägga till en autodelete [policy](https://console.cloud.google.com/firestore/databases/-default-/ttl?project=greenfield-prod-cp8j).

Blir mycket lättre än att behöva hantera tusentals meddelanden sen.

Reply:
Annars ser det episkt ut!

Reply:
Mkt bra idé, vi har lagt in ett `expireAt`-fält nu.